app/server/ruby/lib/sonicpi/lang/core.rb

```diff
@@ -3418,6 +3418,103 @@ def with_random_seed(seed, &block)
+
+      ",
+          examples:       ["
+  use_random_type :white # use white noise as the distribution (default)
+  use_random_seed 1 # reset random seed to 1
+  puts rand # => 0.733917236328125
+  puts rand  #=> 0.464202880859375
+  use_random_seed 1 # reset it back to 1
+  use_random_type :pink # use pink noise as the distribution
+  puts rand # => 0.464202880859375
+  puts rand # => 0.445465087890625
+  use_random_seed 1 # reset it back to 1
+  use_random_type :perlin # use perlin noise as the distribution
+  puts rand # => 0.521209716796875
+  puts rand # => 0.531524658203125
+
+  with_random_type :white do # reset seed back to 1 just for this block
+    puts rand # => 0.733917236328125
+    puts rand #=> 0.464202880859375
+  end
+
+  puts rand # => 0.541961669921875
+            # notice how the last generator (perlin) is restored"]
+
+
+
+
+
+
+      def with_random_type(noise_type, &block)
+        raise ArgumentError, "with_random_type requires a block. Perhaps you meant use_random_type" unless block
+        raise ArgumentError, "invalid noise type '#{noise_type}' - please use one of :white, :pink or :perlin instead" unless %w(white pink perlin).include?(noise_type.to_s)
+        new_thread_gen_type = SonicPi::Core::SPRand.get_random_number_distribution
+
+        SonicPi::Core::SPRand.set_random_number_distribution!(noise_type)
+        res = block.call
+        SonicPi::Core::SPRand.set_random_number_distribution!(new_thread_gen_type)
+        res
+      end
+      doc name:           :with_random_type,
+          introduced:     Version.new(3,2,3),
+          summary:        "Specify random distribution for code block",
+          doc:            "Resets the random number generator to the specified noise type for the specified code block. All generated random numbers and randomisation functions such as `shuffle` and `choose` within the code block will use this new generator. Once the code block has completed, the original generator is restored and the code block generator is discarded. Use this to change the sequence of random numbers in your piece in a way that can be reproduced. Especially useful if combined with iteration. See examples.",
+          args:           [[:noise_type, :symbol]],
+          opts:           nil,
+          accepts_block:  true,
+          requires_block: true,
+          examples:      ["
+  use_random_type :white # use white noise as the distribution (default)
+  use_random_seed 1 # reset random seed to 1
+  puts rand # => 0.733917236328125
+  puts rand  #=> 0.464202880859375
+  use_random_seed 1 # reset it back to 1
+  use_random_type :pink # use pink noise as the distribution
+  puts rand # => 0.464202880859375
+  puts rand # => 0.445465087890625
+  use_random_seed 1 # reset it back to 1
+  use_random_type :perlin # use perlin noise as the distribution
+  puts rand # => 0.521209716796875
+  puts rand # => 0.531524658203125
+
+  with_random_type :white do # reset seed back to 1 just for this block
```

Comment:
I think the comment here is wrong (with_random_type doesn't reset the seed does it)?